lib/release.js

```diff
@@ -0,0 +1,437 @@
+'use strict';
+
+const path = require('path');
+const fs = require('fs').promises;
+const semver = require('semver');
+const replace = require('replace-in-file');
+
+const { getMergedConfig } = require('./config');
+const { runAsync, runSync } = require('./run');
+const { writeJson, readJson } = require('./file');
+
+class Release {
+  constructor(state, argv, cli, req, dir) {
+    this.cli = cli;
+    this.dir = dir;
+    this.newVersion = argv.newVersion;
+    this.isSecurityRelease = argv.security;
+    this.isLTS = false;
+    this.ltsCodename = '';
+    this.date = '';
+    this.config = getMergedConfig(this.dir);
+
+    const { upstream, owner, repo, newVersion } = this;
+
+    this.versionComponents = {
+      major: semver.major(newVersion),
+      minor: semver.minor(newVersion),
+      patch: semver.patch(newVersion)
+    };
+
+    this.stagingBranch = `v${this.versionComponents.major}.x-staging`;
+
+    const upstreamHref = runSync('git', [
+      'config', '--get',
+      `remote.${upstream}.url`]).trim();
+    if (!new RegExp(`${owner}/${repo}(?:.git)?$`).test(upstreamHref)) {
+      cli.warn('Remote repository URL does not point to the expected ' +
+        `repository ${owner}/${repo}`);
+    }
+  }
+
+  async prepare() {
+    const { cli, newVersion, versionComponents } = this;
+
+    // Create new proposal branch.
+    const shouldBranch = await cli.prompt(
+      `Create new proposal branch for ${newVersion}?`, true);
+    if (!shouldBranch) return this.abort();
+    await this.createProposalBranch();
+
+    // Update version and release info in src/node_version.h.
+    const shouldUpdateNodeVersion = await cli.prompt(
+      `Update 'src/node_version.h' for ${newVersion}?`, true);
+    if (!shouldUpdateNodeVersion) return this.abort();
+    await this.updateNodeVersion();
+
+    // Check whether to update NODE_MODULE_VERSION (default false).
+    const shouldUpdateNodeModuleVersion = await cli.prompt(
+      'Update NODE_MODULE_VERSION?', false);
+    if (shouldUpdateNodeModuleVersion) {
+      const runtime = await cli.promptInput(
+        'Specify runtime (ex. \'node\') for new NODE_MODULE_VERSION:',
+        false);
+      const variant = await cli.promptInput(
+        'Specify variant (ex. \'v8_7.9\') for new NODE_MODULE_VERSION:',
+        false);
+      const versions = await cli.promptInput(
+        'Specify versions (ex. \'14.0.0-pre\') for new NODE_MODULE_VERSION:',
+        false);
+      this.updateNodeModuleVersion(runtime, variant, versions);
+    }
+
+    // Update any REPLACEME tags in the docs.
+    const shouldUpdateREPLACEMEs = await cli.prompt(
+      'Update REPLACEME items in docs?', true);
+    if (!shouldUpdateREPLACEMEs) return this.abort();
+    await this.updateREPLACEMEs();
+
+    this.date = await cli.promptInput(
+      'Enter release date in YYYY-MM-DD format:');
+
+    // Update Changelogs
+    const shouldUpdateChangelogs = await cli.prompt(
```

Comment:
Another one that probs should "just happen"™️

Reply:
good point - i can just log that it happened without making it interactive

Reply:
i think for this one i do want to remove the prompt to _start_ editing changelogs, but for similar reasons to the release commit i still would potentially want to pause and confirm that the releaser is happy with the updated changelog before proceeding, so that they can shove in an extra notable commit or something 🤔